src/domain/kaivuilmoitus/components/ApplicationReportOperationalConditionDialog.tsx

```diff
@@ -0,0 +1,192 @@
+}) => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { data: application, refetch } = useApplication(applicationId);
+  const formContext = useForm<ReportOperationalConditionData>({
+    mode: 'onTouched',
+    resolver: yupResolver(reportOperationalConditionSchema),
+    context: {
+      application: application,
+      dateBeforeStartErrorMessageKey: 'dateBeforeStart',
+      dateInFutureErrorMessageKey: 'dateInFuture',
+    },
+  });
+  const { handleSubmit, reset: resetForm, formState } = formContext;
+  const isConsifmButtonEnabled = formState.isValid;
+  const {
+    mutate,
+    reset: resetMutation,
+    isLoading,
+    isError,
+  } = useMutation(reportOperationalCondition);
+  const { showReportOperationalConditionSuccess, showReportOperationalConditionError } =
+    useApplicationReportOperationalConditionNotification();
+  const dialogTitle = t('hakemus:operationalConditionDialog:title');
+  const { id, applicationData, ilmoitukset } = application as Application;
+  const previousReports = ilmoitukset?.TOIMINNALLINEN_KUNTO ?? ([] as Ilmoitus[]);
+  const startDate = applicationData.startTime as Date;
+  const today = new Date();
+
+  useEffect(() => {
+    if (isOpen) {
+      refetch().then(() => {
+        formContext.setValue('applicationId', applicationId);
+      });
+    }
+  }, [isOpen, refetch, formContext, applicationId]);
+
+  function handleClose() {
+    resetForm();
+    resetMutation();
+    onClose();
+  }
+
+  async function submitForm(data: ReportOperationalConditionData) {
+    mutate(data, {
+      onSuccess() {
+        showReportOperationalConditionSuccess();
+        handleClose();
+      },
+      onError() {
+        showReportOperationalConditionError();
+        handleClose();
+      },
+    });
+  }
+
+  return (
+    <Dialog
+      id="application-report-operational-condition"
+      isOpen={isOpen}
+      aria-labelledby={dialogTitle}
+      variant="primary"
+      close={onClose}
+      closeButtonLabelText={t('common:ariaLabels:closeButtonLabelText')}
+    >
+      <Dialog.Header
+        id="application-report-operational-condition-title"
+        title={dialogTitle}
+        iconLeft={<IconInfoCircleFill aria-hidden="true" />}
+      />
+      <FormProvider {...formContext}>
+        <form onSubmit={handleSubmit(submitForm)}>
+          <Dialog.Content>
+            <Instructions previousReports={previousReports} />
+            <VisuallyHiddenInput
+              name="applicationId"
+              value={id as number}
+              aria-hidden="true"
+              readOnly
+            />
+            <DatePicker
+              name="date"
+              label={t(`hakemus:operationalConditionDialog:date`)}
+              locale={locale}
+              minDate={startDate}
+              maxDate={today}
+              initialMonth={today}
+              helperText={t('form:helperTexts:dateInForm')}
+              required
+            />
+            {isError && (
```

Comment:
Not quite sure, it came along when I copy-pasted code from another place. I guess it is not necessary here because error message (of submitting the form and calling the backend) is shown differently. I took this block out.